packages/offix-datastore/src/Model.ts

```diff
@@ -17,6 +18,40 @@ export type Fields<T> = {
     [P in keyof T]: FieldOptions
 };
 
+/**
+ * Model Config options
+ */
+export interface ModelConfig<T = unknown> {
+    /**
+     * Model name
+     */
+    name: string;
+
+    /**
+     * Model store name, defualts to `user_${name}`
+     */
+    storeName?: string;
+
+    /**
+     * Model fields
+     */
+    fields: Fields<T>;
+
+    /**
+     * Delta Query and Subscription filter
+     */
+    predicate?: Predicate<T>;
+
+    /**
+     * Associate server version of entities with local version.
+     * It returns a predicate used to determine which entities
+     * get changed given the data from the server
+     *
+     * @param data is the version from the server
+     */
+    matcher?: (data: T) => Predicate<T>;
```

Comment:
I'm not really getting what this does? Should we simply have two fields instead:

- subscriptionPredicate
- deltaQueryPredicate